### PR TITLE
Make Lua::load load binary chunks when unsafe

### PR DESCRIPTION
**Pull request by HybridEidolon**
This PR makes `Lua::load` also support loading binary chunks with automatic detection when the context is created with `unsafe_new`. It also exposes the `LUA_SIGNATURE` constant for client code to detect whether or not a chunk is text or binary.

I am requesting this for support for a specific application where compiled chunks are the only scripts available.

### CODE REVIEW

**Comment by khvzak** [comment]
Thanks for PR! The ability to load binary chunks indeed a good feature.

A few comments:
I believe the single method `load()` could be used to load binary chunks when Lua created with `unsafe_new()` call. The `LUA_SIGNATURE` constant is exported to check the chunks are binary or text.
Plus it corresponds to the native lua-way to load chunks using a single [`luaL_loadbufferx`](https://www.lua.org/manual/5.3/manual.html#luaL_loadbufferx) function.
To keep API backward compatibility, new methods to `Chunk` could be added, to force `binary` or `text` mode (or auto by default).

Provide different API interfaces for different Lua versions isn't a good idea. Ideally, all code should compile without requiring changes when switching between Lua versions (this is one of `mlua` goals). Plus, we cannot guarantee that Lua >=5.2 are vulnerability-free when executing binary code (if they are not known for now does not mean they does not exists).
This even mentioned in the Lua [manual](https://www.lua.org/manual/5.3/manual.html#pdf-load):

> Lua does not check the consistency of binary chunks. Maliciously crafted binary chunks can crash the interpreter.

**Comment by HybridEidolon** [comment]
So, are you suggesting an interface more like:

1. Create the `Lua` with `unsafe_new`,
2. Being an unsafe context, `load` will now use the "bt" mode instead of exclusively "t", allowing `luaL_loadbufferx` to automatically detect the chunk type passed in,

and having that be the standard way of loading binary chunks through mlua?

Not being able to specify "t" or "b" is the problem that I ran into, because mlua only passes "t" to `loadbufferx`.

**Comment by HybridEidolon** [comment]
I replaced this PR to simply use "bt" mode when the Lua context is unsafe.

File: src/ffi/mod.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ pub use self::lua::{
     lua_Unsigned, lua_Writer,
 };
 
+// Constants
+pub use self::lua::LUA_SIGNATURE;
+
```
Comment (khvzak) [inline src/ffi/mod.rs:41]: For constants there is a dedicated section under comment `constants from lua.h`. This particular constant seems not in use and could be not imported.

File: src/lib.rs [hunk 2]
```diff
@@ -76,7 +76,7 @@ mod userdata;
 mod util;
 mod value;
 
-pub use crate::ffi::lua_State;
+pub use crate::ffi::{lua_State, LUA_SIGNATURE};
```
Comment (khvzak) [inline src/lib.rs:79]: This is an internal constant, and better not to export it publicly

File: tests/function.rs [hunk 3]
```diff
@@ -10,7 +10,7 @@
 )]
 extern "system" {}
 
-use mlua::{Function, Lua, Result, String};
+use mlua::{Function, Lua, Number, Result, String};
```
Comment (khvzak) [inline tests/function.rs:13]: unused

**Pull request merged**